apps/cyberstorm-nextjs/middleware.ts

```diff
@@ -0,0 +1,56 @@
+import { NextResponse } from "next/server";
+import type { NextRequest } from "next/server";
+import {
+  decode,
+  handleJWTGenerationAndReturnResponseOrRedirect,
+  redirectToPremium,
+} from "./utils/auth";
+import { errors } from "jose";
+
+export const config = {
+  matcher: [
+    /*
+     * Match all request paths except for the ones starting with:
+     * - _next/static (static files)
+     * - _next/image (image optimization files)
+     * - favicon.ico (favicon file)
+     */
+    "/((?!_next/static|_next/image|favicon.ico|images).*)",
+  ],
+};
+
+export async function middleware(request: NextRequest) {
+  const req = request;
+  const response = NextResponse.next();
+
+  // For the sake of keeping dev configuration easy and not confusing
+  if (process.env.CYBERSTORM_AUTH_ENABLED) {
+    return response;
+  }
+
+  // JWT exists?
+  if (request.cookies.has("CyberstormAuth")) {
+    // Validate JWT
+
+    const JWTToken = request.cookies.get("CyberstormAuth");
+    if (!JWTToken) {
+      // Token value was empty, redirect to "Go Premium"
+      return redirectToPremium(request, response);
+    }
+
+    try {
+      await decode(JWTToken.value);
+    } catch (e) {
+      if (e instanceof errors.JWTExpired) {
+        // Expired so figure and validate django session and generate a new JWT token
+        return handleJWTGenerationAndReturnResponseOrRedirect(req, response);
+      }
+      throw e;
```

Comment:
If I'm not mistaken the discussion on Discord agreed on redirecting to error page instead of throwing?